### PR TITLE
fix(tab-button): use darker text to pass a11y

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: N/A

---------

- Changes the default Material Design tab color to `$text-color-step-350` or `#a6a6a6`
- Changes the default iOS tab color to `$text-color-step-400` or `#999999`
- Removes the axe skip in the e2e test

These were the minimum color changes needed to pass axe.

### CODE REVIEW

**Review by thetaPC — CHANGES_REQUESTED** [review]
LGTM, but I would like to review the screenshots too. There shouldn't be any issues. I've just learned to not trust screenshots.

**Review by mapsandapps — APPROVED** [review]

File: core/src/themes/ionic.theme.default.md.scss [hunk 1]
```diff
@@ -17,7 +17,7 @@ $overlay-md-background-color:               var(--ion-overlay-background-color,
 $tabbar-md-background:                      var(--ion-tab-bar-background, $background-color) !default;
 $tabbar-md-background-focused:              var(--ion-tab-bar-background-focused, get-color-shade(#fff)) !default;
 $tabbar-md-border-color:                    var(--ion-tab-bar-border-color, var(--ion-border-color, var(--ion-color-step-150, rgba(0, 0, 0, .07)))) !default;
-$tabbar-md-color:                           var(--ion-tab-bar-color, $text-color-step-400) !default;
+$tabbar-md-color:                           var(--ion-tab-bar-color, $text-color-step-350) !default;
```
Comment (sean-perkins) [inline core/src/themes/ionic.theme.default.md.scss:20]: Shouldn't we be seeing some screenshot diffs for MD because of this?

**Review by thetaPC — APPROVED** [review]
LGTM

**Comment by liamdebeasi** [comment]
Discussed with Brandy offline, but I manually removed the MD screenshots in https://github.com/ionic-team/ionic-framework/pull/27355/commits/4c40f185f4460197494a1207698e55760f24963e so they can be regenerated. The diff for the MD screenshots is too small to be detected with our current diff threshold of 0.1. However, I am hesitant to lower the threshold below 0.1 because we risk getting false positives with our screenshots.

**Review by liamdebeasi — APPROVED** [review]
Good to merge once tests pass/screenshots are updated.

**Pull request merged**